code/socialDistribution/decorators.py

```diff
@@ -8,7 +10,8 @@ def unauthenticated_user(view_func):
     """
     def wrapper_func(request, *args, **kwargs):
         if request.user.is_authenticated:
-            return redirect('home')
+            author_id = Author.objects.get(user=request.user).id
```

Comment:
please use pathname incase we change path : `return redirect('socialDistribution:home', author_id=author_id)`
We also might not need the import `from django.http.response import HttpResponseRedirect`
Thanks!